src/phasespace.jl

```diff
@@ -204,21 +207,223 @@ function _clenshaw(L::Int, abs2_2α::Float64, ρ::Matrix{Complex128})
+        data[n+1] = sqrt(binomial(BigInt(N), n)) * (sinth*expphi)^n * (costh*expphi_con)^(N-n)
+    end
+    return result
+end
+
+"""
+    qfuncsu2(ket,Ntheta;Nphi=2Ntheta)
+    qfuncsu2(rho,Ntheta;Nphi=2Ntheta)
+
+Husimi Q SU(2) representation ``⟨θ,ϕ|ρ|θ,ϕ⟩/π`` for the given state. The
+function calculates the SU(2) Husimi representation of a state on the
+generalised bloch sphere (0 < θ < π and 0 < ϕ < 2 π) with a given
+resolution `(Ntheta, Nphi)`.
+
+    qfuncsu2(rho,θ,ϕ)
+    qfuncsu2(ket,θ,ϕ)
+
+This version calculates the Husimi Q SU(2) function at a position given by θ and ϕ.
+"""
+function qfuncsu2(psi::Ket, Ntheta::Int; Nphi::Int=2Ntheta)
+    b = psi.basis
+    psi_bra_data = psi.data'
+    lb = float(b.spinnumber)
+    @assert isa(b, SpinBasis)
+    result = Array{Float64}(Ntheta,Nphi)
+    @inbounds  for i = 0:Ntheta-1, j = 0:Nphi-1
+        result[i+1,j+1] = (2*lb+1)/(4pi)*abs2(psi_bra_data*coherentspinstate(b,pi-i*pi/(Ntheta-1),j*2pi/(Nphi-1)-pi).data)
+    end
+    return result
+end
+
+function qfuncsu2(rho::DenseOperator, Ntheta::Int; Nphi::Int=2Ntheta)
+    b = basis(rho)
+    lb = float(b.spinnumber)
+    @assert isa(b, SpinBasis)
+    result = Array{Float64}(Ntheta,Nphi)
+    @inbounds  for i = 0:Ntheta-1, j = 0:Nphi-1
+        c = coherentspinstate(b,pi-i*1pi/(Ntheta-1),j*2pi/(Nphi-1)-pi)
+        result[i+1,j+1] = abs((2*lb+1)/(4pi)*c.data'*rho.data*c.data)
     end
     return result
 end
 
+function qfuncsu2(psi::Ket, theta::Real, phi::Real)
+    b = basis(psi)
+    psi_bra_data = psi.data'
+    lb = float(b.spinnumber)
+    @assert isa(b, SpinBasis)
+    result = (2*lb+1)/(4pi)*abs2(psi_bra_data*coherentspinstate(b,theta,phi).data)
+    return result
+end
+
+function qfuncsu2(rho::DenseOperator, theta::Real, phi::Real)
+    b = basis(rho)
+    lb = float(b.spinnumber)
+    @assert isa(b, SpinBasis)
+    c = coherentspinstate(b,theta,phi)
+    result = abs((2*lb+1)/(4pi)*c.data'*rho.data*c.data)
+    return result
+end
+
+"""
+    wignersu2(ket,Ntheta;Nphi=2Ntheta)
+    wignersu2(rho,Ntheta;Nphi=2Ntheta)
+
+Wigner SU(2) representation for the given state with a resolution `(Ntheta, Nphi)`.
+The function calculates the SU(2) Wigner representation of a state on the
+generalised bloch sphere (0 < θ < π and 0 < ϕ < 2 π) with a given resolution by
+decomposing the state into the basis of spherical harmonics.
+
+    wignersu2(rho,θ,ϕ)
+    wignersu2(ket,θ,ϕ)
+
+This version calculates the Wigner SU(2) function at a position given by θ and ϕ
+"""
+function wignersu2(rho::DenseOperator, theta::Real, phi::Real)
+
+    N = length(basis(rho))-1
+
+    ### Tensor generation ###
+    BandT = Array{Vector{Float64}}(N,N+1)
+    BandT[1,1] = collect(linspace(-N/2, N/2, N+1))
+    BandT[1,2] = -collect(sqrt.(linspace(1, N, N)).*sqrt.(linspace((N)/2, 1/2, N)))
+    BandT[2,1] = clebschgordan(1,0,1,0,2,0)*BandT[1,1].*BandT[1,1] -
+    clebschgordan(1,-1,1,1,2,0)*append!(zeros(N+1-length(BandT[1,2])),BandT[1,2].*BandT[1,2]) -
```

Comment:
It's cleaner to create the vector here with `[zeros(N+1-length(BandT[1,2])); BandT[1,2].*BandT[1,2])]` (mind the semicolon). Same goes for multiple occasions in this function.